### ISSUE TITLE
Can no longer use two 'highlight feedlist' entries on config

### ISSUE THREAD

**Issue by Rahtgaz** [akrennmair/newsbeuter#568]
I've always used two 'highlight feedlist' entries on config two distinguish between a feed item a totals item on the feedlist. The screenshot below shows that I can no longer do this.

config file:
```
# [...]
highlight feedlist ".+ N .+" color137 default
highlight feedlist ".+ --> .+" color242 default
# [...]
```
Despite the fact that both regex end up overlapping, newsbeuter was smart to apply the second one on top of the first, giving me a dark grey for the totals item and an orange to unread items. This no longer happens.

![screenshot_2017-06-16_18-31-58](https://user-images.githubusercontent.com/10880397/27237733-3f440560-52c2-11e7-97f2-a4d7dd47de84.png)

**Comment by Minoru** [comment]
Thanks for a report!

I tried d1c116d5770, r2.9, r2.8, r2.7 and r2.6, and I couldn't make any of them use two colours—all of them always used the second regex only if the first one didn't match. I suspect the culprit might be not in the Newsbeuter itself but in one of the libraries we rely on.

@Rahtgaz, what was the last version where you saw the desired behaviour? Also, what distribution are you using? Or is it OS X?

**Comment by Rahtgaz** [comment]
Hello Minoru,

Unfortunately I simply cannot remember the version number anymore. It was based of Ubuntu MATE 17.04. According to the ubuntu repos, that would be version 2.9, circa 2015 (https://packages.ubuntu.com/xenial/newsbeuter). The source package to the right points to that year, if you download the orig sources and inspect the CHANGES file.

Last week my old computer finally died and I took the chance to fully move to Arch. Had been runing it for a couple of years on a VM). So now I'm running newsbeuter 2.9-6 of that distro, last built on 2017-03-08.

**Comment by Minoru** [comment]
I managed to reproduce the desired behaviour with 2.8, but only when I swapped the highlight rules.

Highlighting was broken in 2.9 so that only the last rule applied (#116), so naturally nothing worked there.

The latest Git revision,  d1c116d, applies "-->" rule only for unread feeds, and completely ignores the "N" rule.

I give up for now.

**Comment by tsipinakis** [comment]
You probably meant to link to #166 not #116.

The 2.9-3 version in Debian (and Ubuntu, since they pull packages from Debian) has, among others, the fix for #166 backported., so that's probably why highlight worked in Ubuntu but not in Arch.

**Comment by Rahtgaz** [comment]
> I managed to reproduce the desired behaviour with 2.8, but only when I swapped the highlight rules.

Yes, that's my bad. That's the result of me experimenting, trying to get it to work. The rules needs to be reversed, so that the normal behaviour -- read second rule only if first fails -- functions as expected.

**Comment by Minoru** [comment]
> You probably meant to link to #166 not #116.

Yep; thanks!

**Comment by Rahtgaz** [comment]
@Minoru,

Looking at the file history, I think this may be the commit that broke it: https://github.com/akrennmair/newsbeuter/commit/26522a9315d5ee47be0ae2783d32331178eaaa7e#diff-fd4d69e28b478905918fa0b3921b6dce

I'm not sure since I don't remember much about C++. But seems like something broke when you move the strings to use C11 variadic templates.

**Comment by Minoru** [comment]
Cool! To be clear on this point: did you build with and without that patch to verify?

I plan to build 2.9-3 (Debian patched version) some time this week to see how it behaves, and then run `git-bisect` from 2.8 to current master to find out where it was broken. We probably have a lot of changes since then, so fixing this won't be that simple, but at least we'll know what caused the regression.
